### PR TITLE
docs: added cheatsheet for best practices

### PR DESCRIPTION
**Pull request by reemardelarosa**
# What is it?

- [ ] Feature / enhancement
- [ ] Bug
- [x] Docs / tests

# Description

Add Cheatsheet: Best Practices created by Misko

# Use cases and why
Screenshot 2022-12-08 at 5 43 38 PM

# Checklist:

- [x] My code follows the [developer guidelines of this project](https://github.com/BuilderIO/qwik/blob/main/CONTRIBUTING.md)
- [x] I have performed a self-review of my own code
- [x] I have made corresponding changes to the documentation
- [ ] Added new tests to cover the fix / functionality

### CODE REVIEW

**Comment by KenAKAFrosty** [comment]
Hey thanks a ton for doing this!

Also just wanted to point out the "Or Habits...." and "Or Dehydrate...." sentences just under the main header were more internal notes I think? Can probably just leave those off.

**Comment by reemardelarosa** [comment]
@KenAKAFrosty updated

File: packages/docs/src/routes/docs/cheat/anti-patterns/index.mdx [hunk 1]
```diff
@@ -0,0 +1,107 @@
+---
+title: Anti-patterns
+contributors:
+  - mhevery
+---
+
+# No-Hydration Tips
+## Don't register events eagerly with `useClientEffect$()`
```
Comment (wmertens) [inline packages/docs/src/routes/docs/cheat/anti-patterns/index.mdx:8]: How about changing this to `Don't register DOM events eagerly with `useClientEffect$()`? I was confused until I realized it was about DOM

File: packages/docs/src/routes/docs/cheat/anti-patterns/index.mdx [hunk 2]
```diff
@@ -0,0 +1,107 @@
...
+    // Do the thing
+}
+//OR 
+doTheThing(location.query);
+    
+```
+
+
+## EXCEPTION
+When doing SSG for pure static files, this is a necessary evil. The server won't have current location information when the server generates the files at build time.
```
Comment (wmertens) [inline packages/docs/src/routes/docs/cheat/anti-patterns/index.mdx:103]: I don't understand this. `useLocation()` gets the location of the page that is being SSR'ed, no? And then when it runs on the client it gets the actual file location? Doesn't it still work with SSG?

File: packages/docs/src/routes/docs/cheat/anti-patterns/index.mdx (hunk 1 above)
Comment (reemardelarosa) [reply packages/docs/src/routes/docs/cheat/anti-patterns/index.mdx:8]: @KenAKAFrosty

File: packages/docs/src/routes/docs/cheat/anti-patterns/index.mdx (hunk 2 above)
Comment (reemardelarosa) [reply packages/docs/src/routes/docs/cheat/anti-patterns/index.mdx:103]: @KenAKAFrosty

Comment (manucorporat) [reply packages/docs/src/routes/docs/cheat/anti-patterns/index.mdx:103]: yeah, i dont understand this exception either, `useLocation()` will also work in SSG, SSG is the same as SSR (just cached forever)

**Comment by manucorporat** [comment]
I am not sure we should call this Anti-patterns, bit negative? how about `Best practices`?

File: packages/docs/src/routes/docs/cheat/anti-patterns/index.mdx (hunk 2 above)
Comment (reemardelarosa) [reply packages/docs/src/routes/docs/cheat/anti-patterns/index.mdx:103]: @mhevery

File: packages/docs/src/routes/docs/cheat/anti-patterns/index.mdx (hunk 1 above)
Comment (reemardelarosa) [reply packages/docs/src/routes/docs/cheat/anti-patterns/index.mdx:8]: @mhevery

**Comment by reemardelarosa** [comment]
@KenAKAFrosty @mhevery do we have a resolution for the comments here from @manucorporat nad @wmertens ?

**Comment by mhevery** [comment]
Please change it to `Best Practices` and let's get this merged in.

**Comment by reemardelarosa** [comment]
@mhevery @manucorporat @KenAKAFrosty updated to Best Practices

**Comment by reemardelarosa** [comment]
@manucorporat @KenAKAFrosty can we merge this PR or do we need to update more?

cc: @mhevery

**Pull request merged**